### DISCUSSION TITLE
How can I use <use> and SVG in Storybook/Webpack?

### DISCUSSION THREAD

**Discussion by raph90** [storybookjs/storybook#25779] [Help]
### Summary

I'm writing an app using Redwood JS (and Storybook). 

I'm trying to load in SVGs into my page using dynamic URLs, and I think the best way to do this is by using <use> and then using a template string with a prop name to load in the correct symbol from my svg sprite, something like this:

```js
import Icons from 'src/icons/icon-sprite.svg'
...

<svg>
      <use href={`${Icons}#${myPropName}`} />
</svg>
```

(Note - I've seen other ways to do dynamic imports online, but they usually result in SVG in JSX which I want to avoid for bundle size). 

However, the difficulty I'm running into is getting that Icons import to actually work correctly. I have this in my Storybook webpack config: 

```
config.module.rules[0].oneOf.unshift({
    test: /\.svg$/,
    type: 'asset/resource',
    use: [
      {
        loader: 'svg-inline-loader',
      },
    ],
  })
```
That syntax might not seem normal but Redwood adds its own abstraction over the top. At any rate, this rule is getting used. When I do this, the `import Icons` resolves to a file_name.svg. However, that file is not appearing in my Webpack bundle on Storybook, and I'm not sure how to get it to. 

I've used a few different loaders to try and getting this working but I really don't know where to go next with it. I'd really appreciate any help anyone could provide. 

### Additional information

My `storybook.config.js` from Redwood:

```js
const path = require('path')

async function webpackFinal(config) {
  config.resolve.alias = {
    ...config.resolve.alias,
    'original-dayjs': path.resolve(__dirname, '../../node_modules/dayjs'),
    '@redwoodjs/testing/web-original': path.resolve(
      __dirname,
      '../../node_modules/@redwoodjs/testing/web'
    ),
    src: path.resolve(__dirname, '../src'),
  }

  const stylesRules = [
    '\\.css$',
    '\\.module\\.css$',
    '\\.scss$',
    '\\.module\\.scss$',
  ]

  config.module.rules[0].oneOf.forEach((rule) => {
    if (stylesRules.includes(rule.test.source)) {
      rule.use.push('postcss-loader')
    }
  })

  config.module.rules[0].oneOf.unshift({
    test: /\.svg$/,
    type: 'asset/resource',
    use: [
      {
        loader: 'svg-inline-loader',
      },
    ],
  })

  config.module.rules.push({
    test: /\.mdx?$/,
    use: [
      {
        loader: '@mdx-js/loader',
      },
    ],
  })

  return config
}

module.exports = {
  addons: [
    '@storybook/addon-essentials',
    '@storybook/addon-styling',
    'storybook-dark-mode',
    '@storybook/addon-controls',
    // 'storybook-preset-inline-svg',
  ],
  webpackFinal,
}
```

**Comment by valentinpalkovic** [comment]
I think the issue is connected with Storybook also setting up an `asset/resource` loader for svgs, which you first have to filter out. You can take a look [here](https://github.com/storybookjs/storybook/issues/16438#issuecomment-1910195664), where I have recommended something very similar.
